pkg/ingester/activeseries/active_series.go

```diff
@@ -163,30 +163,28 @@ func (s *seriesStripe) updateSeriesTimestamp(now time.Time, series labels.Labels
 	}
 }
 
-func (s *seriesStripe) findEntryForSeries(fingerprint uint64, series labels.Labels) *atomic.Int64 {
+func (s *seriesStripe) findEntryForSeries(ref uint64) *atomic.Int64 {
 	s.mu.RLock()
 	defer s.mu.RUnlock()
 
 	// Check if already exists within the entries.
-	for _, entry := range s.refs[fingerprint] {
-		if labels.Equal(entry.lbs, series) {
-			return entry.nanos
-		}
+	entry, ok := s.refs[ref]
+	if !ok {
+		return nil
 	}
 
-	return nil
+	return entry.nanos
```

Comment:
All of this can be just:
```go
return s.refs[ref].nanos
```
(Which works for zero valued seriesEntry returning nil)